concordium-consensus/src/Concordium/External.hs

```diff
@@ -1131,22 +1131,37 @@ checkIfWeAreFinalizer cptr = do
     res <- runMVR Q.checkIsCurrentFinalizer mvr
     return $! if res then 1 else 0
 
--- |Check whether we are a baker from the perspective of the best block.
--- Returns -1 if we are not added as a baker.
--- Returns -2 if we are added as a baker, but not part of the baking committee yet.
--- Returns -3 if we have keys that do not match the baker's public keys on the chain.
--- Returns >= 0 if we are part of the baking committee. The return value is the
--- baker id as appearing in blocks.
-bakerIdBestBlock :: StablePtr ConsensusRunner -> IO Int64
-bakerIdBestBlock cptr = do
+-- |Check if consensus is running.
+-- Returns 0 for 'False' and 1 for 'True'.
+checkIfRunning :: StablePtr ConsensusRunner -> IO Word8
+checkIfRunning cptr = do
     (ConsensusRunner mvr) <- deRefStablePtr cptr
-    res <- runMVR Q.getBakerStatusBestBlock mvr
-    return $! case res of
-        NoBaker -> -1
-        InactiveBaker _ -> -2
-        BadKeys _ -> -3
-        ActiveBaker bid -> fromIntegral bid
-
+    res <- runMVR Q.checkIsShutDown mvr
+    return $! if res then 0 else 1
+
+-- |Check whether we are a baker from the perspective of the best block. 
+-- Puts the baker ID into the ptr "bakerIdPtr" along with a flag that a baker ID is found into "hasBakerIdPtr"
+-- Returns 1 if we are not added as a baker.
+-- Returns 2 if we are added as a baker, but not part of the baking committee yet.
+-- Returns 3 if we have keys that do not match the baker's public keys on the chain.
+-- Returns 0 if we are part of the baking committee. 
+bakerStatusBestBlock :: StablePtr ConsensusRunner -> Ptr Word64 -> Ptr Word8 -> IO Word8
+bakerStatusBestBlock cptr bakerIdPtr hasBakerIdPtr = do
+    (ConsensusRunner mvr) <- deRefStablePtr cptr
+    (bs, mBid) <- runMVR Q.getBakerStatusBestBlock mvr
+    case mBid of
+        Nothing -> return $! getBakerStatusCode bs
```

Comment:
I think it would also be wise to `poke hasBakerIdPtr 0` here. Also document the meaning of this value.